app/goprotoc/codegen.go

```diff
@@ -38,61 +39,150 @@ var protocOutputs = map[string]struct{}{
 	"ruby":     {},
 }
 
+type outputLocation struct {
+	path         string
+	isZip, isJar bool
+}
+
+type outputFile struct {
+	outputLocation
+	fileName string
+}
+
+func (f outputFile) String() string {
+	if f.isZip || f.isJar {
+		return fmt.Sprintf("%s:%s", f.path, f.fileName)
```

Comment:
Is this really necessary? This is almost more confusing to me than just having the path

Reply:
This is when it is trying to convey the name of a file that is *in* an archive. So just using the path will make it look like the archive is just a directory.

Reply:
But this path will end in `.jar` or `.zip` right? Doesn't this convey that well enough? :-)

Reply:
Nope. `f.path` ends in `.jar` or `.zip`. The part after the colon is the name of a file _in_ that archive.

Reply:
Ah got it, carry on

Reply:
I'll add some doc comments to the type to better explain what this is and how it differs from `outputLocation`.

Reply:
I should probably do a closer review, this was just a first pass :-)